app/src/main/java/com/arctouch/codechallenge/home/fragments/MovieListFragment.kt

```diff
@@ -0,0 +1,55 @@
+package com.arctouch.codechallenge.home.fragments
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProviders
+import com.arctouch.codechallenge.R
+import com.arctouch.codechallenge.api.TmdbApi
+import com.arctouch.codechallenge.home.adapters.HomeAdapter
+import com.arctouch.codechallenge.viewmodel.MovieListViewModel
+import com.arctouch.codechallenge.viewmodel.MovieListViewModelFactory
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.schedulers.Schedulers
+import kotlinx.android.synthetic.main.movie_list_fragment.*
+
+class MovieListFragment : Fragment() {
+    private lateinit var moviesViewModel: MovieListViewModel
+    private lateinit var moviesViewModelFactory: MovieListViewModelFactory
+    private lateinit var moviesAdapter: HomeAdapter
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        return inflater.inflate(R.layout.movie_list_fragment, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        prepareRecyclerView()
+        loadFactories()
+        loadViewModels()
+    }
+
+    private fun prepareRecyclerView() {
+        moviesAdapter = HomeAdapter()
+        recyclerView.apply {
+            adapter = moviesAdapter
+        }
+    }
+
+    private fun loadFactories() {
+        moviesViewModelFactory = MovieListViewModelFactory(TmdbApi.create())
+    }
+
+    private fun loadViewModels() {
```

Comment:
Do you think that this function are align with single responsibility principle? or his name is misleading?

Reply:
I'm not just loading the ViewModels, I'm also creating the observables, so it's not really aligned with single responsibility principle. It can be solved by renaming this method, separating the load of the ViewModel from the creation of the observable, or just renaming to something like "createObservables" and remove the unnecessary class attributes (ViewModels), since it's only used here.